components/GraphDisplays/monthlyExpenses.tsx

```diff
@@ -0,0 +1,228 @@
+import { AntDesign } from "@expo/vector-icons";
+import React, { useEffect, useState } from "react";
+import { View, Text } from "react-native";
+import { TouchableHighlight } from "react-native-gesture-handler";
+import { EventCallbackInterface, StringOrNumberOrList } from "victory-core";
+import { VictoryAxis, VictoryBar, VictoryChart, VictoryTheme } from "victory-native";
+
+interface MonthlyDatum {
+    month: string,
+    year: number,
+    amount: number,
+    id: number,
+}
+
+interface MonthlyExpenseGraphProps {
+    data: MonthlyDatum[]
+    monthSelectedCallback?: (datum: MonthlyDatum) => null,
+    mainColour?: string,
+    highlightColour?: string,
+}
+
+interface ExteralMutation {
+    target: string,
+    eventKey: string,
+    mutation: Function,
+    callback: Function
+}
+
+const dumpy = [
+    { year: 2020, amount: 13000, month: "JAN", id: 1 },
+    { year: 2020, amount: 16500, month: "FEB", id: 2 },
+    { year: 2020, amount: 14250, month: "MAR", id: 3 },
+    { year: 2020, amount: 19000, month: "APR", id: 4 },
+    { year: 2020, amount: 16000, month: "MAY", id: 5 },
+    { year: 2020, amount: 14000, month: "JUN", id: 6 },
+    { year: 2020, amount: 12000, month: "JUL", id: 7 },
+    { year: 2020, amount: 16000, month: "AUG", id: 8 },
+    { year: 2020, amount: 14000, month: "SEP", id: 9 },
+    { year: 2020, amount: 17000, month: "OCT", id: 10 },
+    { year: 2020, amount: 21000, month: "NOV", id: 11 },
+    { year: 2020, amount: 19000, month: "DEC", id: 12 },
+];
+
+const MonthlyExpenseGraph = ({ data, monthSelectedCallback, mainColour, highlightColour }: MonthlyExpenseGraphProps) => {
+
+    const [mutations, setMutations] = useState<Array<ExteralMutation>>([])
+    const [selectedMonth, setSelectedMonth] = useState(undefined)
+
+    useEffect(() => {
+        handleClick(selectedMonth ? selectedMonth : -1)
+    }, [data])
+
+    const handleClick = (id: number) => {
+        setMutations([
+            {
+                target: "data",
+                eventKey: "all",
+                mutation: (props: any) => {
+                    if (id == props.datum.id) {
+                        return {
+                            barRatio: 1,
+                            label: 'hi',
+                            style: {
+                                fill: highlightColour ? highlightColour : "#3dbf60",
+                            }
+                        }
+                    } else {
+                        return {
+                            barRatio: 0.5,
+                            style: {
+                                fill: highlightColour ? highlightColour : "#2e8f48"
+                            }
+                        }
+                    }
+                },
+                callback: () => {
+                    if (mutations !== []) {
```

Comment:
These will always return true because js compares objects by reference, not value. You probably mean `mutations.length() === 0` instead.